src/views/clusteroverview/overview/components/details-card/SourceMissingStatus/UpgradeApprovalLink.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+
+import { modelToGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { K8sModel } from '@openshift-console/dynamic-plugin-sdk';
+
+import { InstallPlanModel } from '../utils/models';
+import { SubscriptionKind } from '../utils/types';
+
+import BlueArrowCircleUpIcon from './BlueArrowCircleUpIcon';
+
+const resourcePathFromModel = (model: K8sModel, name?: string, namespace?: string) => {
+  const { plural, namespaced, crd } = model;
+
+  let url = '/k8s/';
+
+  if (!namespaced) {
+    url += 'cluster/';
+  }
+
+  if (namespaced) {
+    url += namespace ? `ns/${namespace}/` : 'all-namespaces/';
+  }
```

Comment:
```suggestion
  if (namespaced) {
    url += namespace ? `ns/${namespace}/` : 'all-namespaces/';
  } else {
    url += 'cluster/';
  }
```
What do you think?

Reply:
This is code I copied from console. I'm working on a PR to expose this components along with other code, so this will eventually be deleted. If I were keeping it, I'd definitely change it to what you have.